SCons/Tool/link.py

```diff
@@ -243,17 +235,14 @@ def _versioned_lib_symlinks(env, libnode, version, prefix, suffix, name_func, so
 
 def _versioned_shlib_symlinks(env, libnode, version, prefix, suffix):
     name_func = env['LINKCALLBACKS']['VersionedShLibName']
-    soname_func = env['LINKCALLBACKS']['VersionedShLibSoname']
+    soname_func = env['ShLibSonameGenerator']
```

Comment:
Why change this?

Reply:
ShLibSonameGenerator ends up calling that linkcallback if SONAME is not supplied. Without this change, the ShLibSonameGenerator is skipped out.